### ISSUE TITLE
codegen fails to generate code for non @model dependencies 

### ISSUE THREAD

**Issue by sebsto** [aws-amplify/amplify-ios#240]
When using a model as :

```graphql
type Coordinate
{
    longitude: Float
    latitude: Float
}


type Landmark
    @model
    @auth(rules: [ {allow: private, provider: userPools, operations: [ read ] } ])
{
    id: ID!
    name: String!
    category: String
    city: String
    state: String
    isFeatured: Boolean
    isFavorite: Boolean
    park: String
    coordinates: Coordinate 
    imageName: String
}
```

Codegen fails with 

```error
Unknown type Coordinate for field coordinates. Did you forget to add the @model directive
Error: Unknown type Coordinate for field coordinates. Did you forget to add the @model directive
    at AppSyncSwiftVisitor.getNativeType (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/amplify-codegen-appsync-model-plugin/src/visitors/appsync-visitor.ts:298:13)
    at /usr/local/lib/node_modules/@aws-amplify/cli/node_modules/amplify-codegen-appsync-model-plugin/src/visitors/appsync-swift-visitor.ts:33:32
    at Array.forEach (<anonymous>)
    at /usr/local/lib/node_modules/@aws-amplify/cli/node_modules/amplify-codegen-appsync-model-plugin/src/visitors/appsync-swift-visitor.ts:32:34
    at Array.forEach (<anonymous>)
    at AppSyncSwiftVisitor.generateStruct (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/amplify-codegen-appsync-model-plugin/src/visitors/appsync-swift-visitor.ts:27:46)
    at AppSyncSwiftVisitor.generate (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/amplify-codegen-appsync-model-plugin/src/visitors/appsync-swift-visitor.ts:21:22)
```

As a workaround, I added @model on Coordinate and a @connection annotation on Landmark.  This is NOT want I want to model.  I do not want Coordinate to be stored in a DynamoDB table, this is just a type definition.  In previous version Coordinate was a JSON object stored as a property into the Landmark table.  I want to keep this behaviour.

**Comment by Soek22** [comment]
Hello @sebsto ,

1. This has been marked as `BUG` regarding `API` here: https://github.com/aws-amplify/amplify-cli/issues/3026 .

2. Regarding `Datastore` it has been marked as pending `enhancement` here: https://github.com/aws-amplify/amplify-cli/issues/2988 .

Once the second one is implemented, the first one will get resolved.

**Comment by palpatim** [comment]
Related to the various "enum" issues as well.

**Comment by drochetti** [comment]
Hi @sebsto 

Support for non-model types just landed in `master`. We will release a new version of Amplify soon along with a new version of the codegen provided by Amplify CLI, stay tuned. In the meantime if you want to try feel free to use the `master` branch.

I'll keep this issue open until we officially release.

**Comment by sebsto** [comment]
great - will test the new version

**Comment by drochetti** [comment]
@sebsto I believe this issue is fixed in `1.0.0`. Feel free to re-open in case you're still experiencing issues.

**Comment by sebsto** [comment]
I confirm this is working with 1.0.0
